### PR TITLE
fix: add support for ancestry

### PR DESCRIPTION
**Pull request by peternhale**
@W-10833165@

### CODE REVIEW

File: src/package/packageAncestry.ts [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+    }
+    this.display(this.options ? this.options['logger'] : undefined);
+  }
+
+  private createLabel(node: AncestryTreeProducer): string {
+    const subscriberId =
+      this.verbose && node?.options?.node?.SubscriberPackageVersionId
+        ? ` (${node.options.node.SubscriberPackageVersionId})`
+        : '';
+    const label = node?.label ? `${node.label}${subscriberId}` : 'root';
```
Comment (WillieRuemmele) [inline src/package/packageAncestry.ts:343]: NIT: could be inline `return node?.label ? ...`

File: src/package/packageAncestry.ts [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+        producer.children.forEach((child) => dotLines.push(AncestryDotProducer.buildDotEdge(producer, child)));
+      }
+      producers.push(...producer.children);
+    }
+    return `strict graph G {\n${dotLines.join('\n')}}`;
+  }
+}
+
+export class PackageAncestryNode extends AsyncCreatable<PackageAncestryNodeOptions> implements Attributes {
+  private readonly _version: VersionNumber;
```
Comment (WillieRuemmele) [inline src/package/packageAncestry.ts:446]: do we want to keep all of the underscores on private attributes?

File: test/package/packageTest.nut.ts [hunk 3]
```diff
@@ -25,6 +26,13 @@ import { deletePackage } from '../../src/package';
 import { PackageVersion } from '../../src/package';
 import { Package } from '../../src/package';
 import { PackagingSObjects } from '../../src/interfaces';
+import { VersionNumber } from '../../src/utils/versionNumber';
+import { AncestryJsonProducer, PackageAncestry } from '../../src/package/packageAncestry';
+import { AncestryTreeProducer } from '../../src/package/packageAncestry';
+import {
+  AncestryRepresentationProducer,
```
Comment (WillieRuemmele) [inline test/package/packageTest.nut.ts:33]: NIT: should be able to import all of these from `../../src/package` or `/src/interfaces` and combine a few above

**Review by WillieRuemmele — CHANGES_REQUESTED** [review]
empty `package-ancestry.md` file

File: src/package/packageAncestry.ts (hunk 2 above)
Comment (peternhale) [reply src/package/packageAncestry.ts:446]: What do you suggest as an alternative?

Comment (WillieRuemmele) [reply src/package/packageAncestry.ts]: I haven't been prefixing private methods or attributes with anything, just letting the `private` scope take care of it. 🤷‍♂️ either way

**Comment by WillieRuemmele** [comment]
@peternhale I'm still seeing an empty `messages/package-ancestry.md` file, did you want to keep that for something?

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**